ReactNativeClient/lib/reducer.js

```diff
@@ -723,7 +868,7 @@ const reducer = (state = defaultState, action) => {
 
 		case 'SEARCH_UPDATE':
 			{
-				newState = Object.assign({}, state);
+				newState = handleHistory(state, action);
```

Comment:
@laurent22 I forgot to delete this line which is no longer necessary. (handleHistory has been moved outside the switch statement).

Please remove it.